docroot/modules/contrib/videojs/src/Plugin/Field/FieldFormatter/TextLinkFormatter.php

```diff
@@ -0,0 +1,222 @@
+      '#required' => TRUE,
+    ];
+    $element['controls'] = [
+      '#title' => t('Show controls'),
+      '#type' => 'checkbox',
+      '#default_value' => $this->getSetting('controls'),
+    ];
+    $element['autoplay'] = [
+      '#title' => t('Autoplay'),
+      '#type' => 'checkbox',
+      '#default_value' => $this->getSetting('autoplay'),
+    ];
+    $element['loop'] = [
+      '#title' => t('Loop'),
+      '#type' => 'checkbox',
+      '#default_value' => $this->getSetting('loop'),
+    ];
+    $element['muted'] = [
+      '#title' => t('Muted'),
+      '#type' => 'checkbox',
+      '#default_value' => $this->getSetting('muted'),
+    ];
+    $element['annotations'] = [
+      '#title' => t('Enable Annotations'),
+      '#type' => 'checkbox',
+      '#default_value' => $this->getSetting('annotations'),
+    ];
+    $element['extensions'] = [
+      '#title' => t('Allow Video Formats'),
+      '#type' => 'textarea',
+      '#default_value' => $this->getSetting('extensions'),
+    ];
+    $element['preload'] = [
+      '#title' => t('Preload'),
+      '#type' => 'select',
+      '#default_value' => $this->getSetting('preload'),
+      '#options' => array(
+        'none' => 'none',
+        'metadata' => 'metadata',
+        'auto' => 'auto',
+      ),
+      '#description' => t('Hint to the browser about whether optimistic downloading of the video itself or its metadata is considered worthwhile.'),
+    ];
+    return $element;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+
+    $summary = array();
+    $summary[] = t('HTML5 Video (@widthx@height@controls@autoplay@loop@muted@annotations@extensions).', [
+      '@width' => $this->getSetting('width'),
+      '@height' => $this->getSetting('height'),
+      '@controls' => $this->getSetting('controls') ? t(', controls') : '' ,
+      '@autoplay' => $this->getSetting('autoplay') ? t(', autoplaying') : '' ,
+      '@loop' => $this->getSetting('loop') ? t(', looping') : '' ,
+      '@muted' => $this->getSetting('muted') ? t(', muted') : '',
+      '@annotations' => $this->getSetting('annotations') ? t(', annotations') : '',
+      '@extensions' => $this->getSetting('extensions') ? t(', extensions') : '',
+      
+    ]);
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(FieldItemListInterface $items, $langcode) {
+    $elements = array();
+    $video_items = array();
+    $ext = 'mp4';
+    foreach ($items as $delta => $item) {
+      $video_uri = $item->getValue();
+      $video_items[] = Url::fromUri(file_create_url($video_uri['value']));
+      $var = $this->getSettings();
+      $extract_extensions = explode(',', $var['extensions']);
+      foreach ($extract_extensions as $key => $value) {
+        if (strpos($video_uri['value'], $value)) {
+          $ext = $value;
+        }
+        else {
+          $ext = 'mp4';
+        }
+      }
+    }
+    if ($this->getSetting('annotations') == 0) {
```

Comment:
Only change in the 2 arrays below are library. Can we make sure only they are altered conditionally.